src/copy_tree.rs

```diff
@@ -38,38 +40,71 @@ pub fn copy_tree<ST: ReadTree, DT: WriteTree>(
         // deleted or changed while this is running.
         ui::set_bytes_total(source.size()?.file_bytes);
     }
+
+    // Target selected for copy (subpaths from here and included)
+    let target = &options.restore_only;
+    let target_tree: Vec<&str> = target.split('/').collect();
+
     ui::set_progress_phase("Copying");
     for entry in source.iter_entries()? {
-        if options.print_filenames {
-            crate::ui::println(entry.apath());
-        }
-        ui::set_progress_file(entry.apath());
-        if let Err(e) = match entry.kind() {
-            Kind::Dir => {
-                stats.directories += 1;
-                dest.copy_dir(&entry)
-            }
-            Kind::File => {
-                stats.files += 1;
-                dest.copy_file(&entry, source).map(|s| stats += s)
+        // Check if this entry is selected for copy
+        let subtree: Vec<&str> = entry.apath().split('/').collect();
+        let mut to_be_copied: bool = false;
+        
+        match target.as_ref() {
+            "" => to_be_copied = true,
+            _ => {
+                // Take the top path from target and match it with entry (accept all subpaths)
+                if subtree.len() > target_tree.len() {
+                    let mut matched: usize = 0;                    
+                    for (i, _) in target_tree.iter().enumerate() {
+                        if target_tree[i].eq(subtree[i+1]) {
+                            matched = matched + 1; 
+                        }
+                    }
+                    to_be_copied = matched == target_tree.len();
+                }
             }
-            Kind::Symlink => {
-                stats.symlinks += 1;
-                dest.copy_symlink(&entry)
-            }
-            Kind::Unknown => {
-                stats.unknown_kind += 1;
-                // TODO: Perhaps eventually we could backup and restore pipes,
-                // sockets, etc. Or at least count them. For now, silently skip.
-                // https://github.com/sourcefrog/conserve/issues/82
-                continue;
-            }
-        } {
-            ui::show_error(&e);
-            stats.errors += 1;
-            continue;
         }
-        ui::increment_bytes_done(entry.size().unwrap_or(0));
+
+        match to_be_copied {
+            true => {
```

Comment:
Rather than `match { true => { ...` I think you could just say `if to_be_copied {...}`.